sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/MaxBundleGasPerBlockTransactionSelector.java

```diff
@@ -60,15 +49,67 @@ public TransactionSelectionResult evaluateTransactionPostProcessing(
       final TransactionEvaluationContext txContext,
       final TransactionProcessingResult transactionProcessingResult) {
 
-    // short circuit if we are not a PendingBundleTx
-    if (!(txContext.getPendingTransaction() instanceof LineaSendBundle.PendingBundleTx)) {
-      return SELECTED;
-    }
+    if (txContext.getPendingTransaction()
+        instanceof BundlePoolService.TransactionBundle.PendingBundleTx pendingBundleTx) {
+
+      final long gasUsedByTransaction =
+          transactionProcessingResult.getEstimateGasUsedByTransaction();
 
-    final long newCumulativeBundleGas =
-        getWorkingState() + txContext.getPendingTransaction().getTransaction().getGasLimit();
-    setWorkingState(newCumulativeBundleGas);
+      final long currentBundleGasUsed =
+          pendingBundleTx.isBundleStart() ? 0L : getWorkingState().currentBundleGasUsed();
+      final long newCurrentBundleGasUsed = currentBundleGasUsed + gasUsedByTransaction;
 
+      final long cumulativeBlockBundleGasUsed = getWorkingState().cumulativeBlockBundleGasUsed();
+      final long newCumulativeBlockBundleGasUsed =
+          cumulativeBlockBundleGasUsed + gasUsedByTransaction;
+
+      setWorkingState(
```

Comment:
This doesn't appear to be thread safe.  

I don't think this is blocking since IIRC we are not  doing parallel block construction, but making this thread-safe or at least adding a getAndSet to AbstractStatefulPluginTransactionSelector would send the right signal to "future us".

Reply:
there are a bunch of things to refactor for parallel block construction, adding this to the list :)